p2p/protocol/ping/ping.go

```diff
@@ -84,14 +102,24 @@ func (ps *PingService) Ping(ctx context.Context, p peer.ID) <-chan Result {
 // Ping pings the remote peer until the context is canceled, returning a stream
 // of RTTs or errors.
 func Ping(ctx context.Context, h host.Host, p peer.ID) <-chan Result {
-	s, err := h.NewStream(network.WithUseTransient(ctx, "ping"), p, ID)
-	if err != nil {
+	fail := func(err error) chan Result {
```

Comment:
Can we do this by introducing a `ping(...) error` function? I find that easier to read than a `fail` function assigned to a variable.

Reply:
well, it is supposed to return a channel so we can't avoid the ugliness -- but we can make it a standalone function.

Reply:
done.